### ISSUE TITLE
CSS Positioning Persists

### ISSUE THREAD

**Issue by egyptianbman** [Polymer/paper-dialog#37]
`paper-dialog` seems to persist the "top" value generated on open. This causes reuse of a dialog with different data to no longer be vertically centered.

Consider a scenario where a dialog may have an image for some items, but not an image for others.  This may cause the dialog to be much higher for the items with images than the dialog for the items without an image.

(without image)
![image](https://cloud.githubusercontent.com/assets/323600/5057140/ee6c0620-6c72-11e4-8c14-f777ddd86c90.png)

(with image)
![image](https://cloud.githubusercontent.com/assets/323600/5057147/fe7a644e-6c72-11e4-8400-6e316fb70daa.png)

(with image, after reloading -- without having selected the "add" button first)
![image](https://cloud.githubusercontent.com/assets/323600/5057150/213fc5be-6c73-11e4-86f8-983004fcf6d6.png)

Looking through the documentation, I couldn't find any way to "reset" the positioning.  Please advise if there is; thanks!

**Comment by FiatPax** [comment]
@egyptianbman Hi, would you please give some more detail on how you fixed it because I'm having the same problem and I don't know where to add your last Javascript code.

**Comment by bittenbytailfly** [comment]
This only seems to be an issue if you are putting dynamic content into the dialog, presumably because it doesn't know the dimensions in advance.

I've tried your fix @morethanreal - but on the latest version (0.5.4), it doesn't seem to do anything.

**Comment by FiatPax** [comment]
Yeah, dynamic content, only on second load of the same element is correctly shown.

**Comment by bittenbytailfly** [comment]
After a bit of searching, I've just resolved this (at least in my app).

It seems the position is wrong because the DOM is changing after the dialog opens - this is easily corrected with the async command, which ensures the dialog opens after the DOM has changed.

It's described here:

https://www.polymer-project.org/docs/polymer/polymer.html

See my latest commit on my project for an example:

https://github.com/bittenbytailfly/deadbolt-password-generator-chrome-mobile-app/commit/3e1f5db818af47bd6843d612c4d4d95a2fc59268

Hope this is of some help to somebody.

**Comment by FiatPax** [comment]
@bittenbytailfly Very well done, I resolved my issue with your way too, thanks for sharing.

**Comment by lucyhe** [comment]
@bittenbytailfly thank you so much, this was exactly what I was looking for.
